examples/code/command-line-parsing/cal_add_interactive2/cal.ml

```diff
@@ -0,0 +1,35 @@
+open Core
+
+let add_days base days =
+  Date.add_days base days
+  |> Date.to_string
+  |> print_endline
+
+let prompt_for_string name of_string =
+  printf "enter %s: %!" name;
+  match In_channel.(input_line stdin) with
+  | None -> failwith "no value entered. aborting."
+  | Some line -> of_string line
+
+[@@@part "1"];;
+let anon_prompt name of_string =
+  let arg = Command.Arg_type.create of_string in
+  Command.Let_syntax.(
+    let%map_open value = anon (maybe (name %: arg)) in
+    match value with
```

Comment:
More succinct, but wrong. The default argument is evaluated eagerly, which is not the desired outcome in this case, since prompt_for_string operates by side effect.

Reply:
erk, good point!